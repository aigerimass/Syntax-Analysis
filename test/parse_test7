Func_with_While(times){
    a = 0;
    rat = 2;
    while(a < times){
        if (a >= 1) {
            d = Read();
        };
        a = a + 1;
    };
}

Main () {
    Func_with_While(Read());
}
