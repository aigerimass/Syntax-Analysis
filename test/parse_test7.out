----------SYNTAX-ANALYSIS----------
Function Definition: [
name = "Func_with_While",
args: times
body:
>Bind(Var "a", 0);

>Bind(Var "rat", 2);

>While(
cond: 
Lt(Var "a", Var "times")
body:
>If(
cond:
Geq(Var "a", 1)
body:
>Bind(Var "d", >Call(name: "Read", args: ));
);
>Bind(Var "a", Plus(Var "a", 1));
);

]
--------
Function Definition: [
name = "Main",
args: 
body:
>Call(name: "Func_with_While", args: >Call(name: "Read", args: ));

]
--------
----------Run----------------------
----------INTERVAL-ANALYSIS--------
Func_with_While (times) {
#times = [1, 1]
a = 0;
#times = [1, 1];
#a = [0, 0];
rat = 2;
#times = [1, 1];
#a = [0, 0];
#rat = [2, 2];
while (a < times) {
if  (a >= 1) {
d = Read ();
};
#times = [1, 1];
#a = [0, 1];
#rat = [2, 2];
a = a + 1;
#times = [1, 1];
#a = [0, 1];
#rat = [2, 2];
};
#times = [1, 1];
#a = [0, 1];
#rat = [2, 2];
};
------------
Main () {
Func_with_While (Read ());
};
------------

